x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityCrossClusterApiKeySigningIT.java

```diff
@@ -212,9 +220,19 @@ private void assertCrossClusterSearchSuccessfulWithoutResult() throws IOExceptio
     }
 
     private void assertCrossClusterSearchSuccessfulWithResult() throws IOException {
+        final long startTimeMillis = System.currentTimeMillis();
         boolean alsoSearchLocally = randomBoolean();
         final Response response = simpleCrossClusterSearch(alsoSearchLocally);
         assertOK(response);
+
+        try {
+            // Call the generic assertion function with the expected 'authentication_success' event
+            assertAuditLogContainsNewEvent(startTimeMillis, "authentication_success");
+        } catch (Exception e) {
+            // Re-throw the exception so the test fails if the log isn't found
+            throw new RuntimeException("Audit log assertion failed after client-side success check passed.", e);
```

Comment:
I think this is a little confusing because I don't think the `AssertionError` thrown by `assertAuditLogContainsNewEvent` would be caught here (since it doesn't extend `Exception`). This would catch the `IOException` thrown by `Stream.readAllLines`. If `readAllLines` fail, I think failing the test with `fail(e, msg)` is better.